--- ./src/descrambler/cwc.c.orig	2014-07-13 01:51:06.000000000 +0800
+++ ./src/descrambler/cwc.c	2014-07-29 15:22:40.000000000 +0800
@@ -285,7 +285,7 @@
 void cwc_emm_nds(cwc_t *cwc, struct cs_card_data *pcard, const uint8_t *data, int len);
 void cwc_emm_cryptoworks(cwc_t *cwc, struct cs_card_data *pcard, const uint8_t *data, int len);
 void cwc_emm_bulcrypt(cwc_t *cwc, struct cs_card_data *pcard, const uint8_t *data, int len);
-
+void cwc_emm_streamguard(cwc_t *cwc, struct cs_card_data *pcard, const uint8_t *data, int len);
 
 /**
  *
@@ -1328,6 +1328,9 @@
       case CARD_BULCRYPT:
         cwc_emm_bulcrypt(cwc, pcard, data, len);
         break;
+      case CARD_STREAMGUARD:
+        cwc_emm_streamguard(cwc, pcard, data, len);
+        break;
       case CARD_UNKNOWN:
         break;
     }
@@ -1836,6 +1839,36 @@
     cwc_send_msg(cwc, data, len, 0, 1, 0, 0);
 }
 
+/**
+ * streamguard emm handler
+ */
+void
+cwc_emm_streamguard(cwc_t *cwc, struct cs_card_data *pcard, const uint8_t *data, int len)
+{
+    //todo
+    tvhlog(LOG_INFO, "cwc", "cwc_emm_streamguard streamguard card data emm get,here lots of works todo...");
+    int match = 0;
+    
+    if (data[0] == 0x87) {
+        if (memcmp(&data[3], &pcard->cwc_ua[4], 4) == 0) {
+            match = 1;
+        }
+    }
+    else if (data[0] == 0x86) {
+        int i;
+        for (i=0; i < pcard->cwc_num_providers; i++) {
+            if (memcmp(&data[40], &pcard->cwc_providers[i].sa[4], 4) == 0) {
+                /*      if (memcmp(&data[3], &cwc->cwc_providers[i].sa[4], 1) == 0) { */
+                match = 1;
+                break;
+            }
+        }
+    }
+    
+    if (match)
+        cwc_send_msg(cwc, data, len, 0, 1, 0, 0);
+}
+
 void
 cwc_emm_cryptoworks(cwc_t *cwc, struct cs_card_data *pcard, const uint8_t *data, int len)
 {
