--- ./src/input/mpegts/dvb_psi.c.orig	2014-07-25 11:28:09.000000000 +0800
+++ ./src/input/mpegts/dvb_psi.c	2014-07-29 15:34:31.000000000 +0800
@@ -1204,11 +1204,13 @@
       i += nanolen;
     }
     break;
-  case 0x4a00://DRECrypt
-    if (caid != 0x4aee) { // Bulcrypt
-      provid = size < 4 ? 0 : buffer[4];
-      break;
+  case 0x4a00:
+    if (caid == 0x4ad2)//streamguard
+       provid=0;
+    if (caid != 0x4aee && caid != 0x4ad2) { // Bulcrypt
+       provid = size < 4 ? 0 : buffer[4];
     }
+    break;
   default:
     provid = 0;
     break;
@@ -1369,7 +1371,8 @@
     case 0x04:
       hts_stream_type = SCT_MPEG2AUDIO;
       break;
-
+      
+    case 0x06: //0x06 is Chinese Cable TV ac3 audio track
     case 0x81:
       hts_stream_type = SCT_AC3;
       break;
