--- support/configure.inc.orig	2013-11-20 17:19:00.000000000 +0800
+++ support/configure.inc	2013-11-24 22:13:52.000000000 +0800
@@ -263,9 +263,10 @@
 #endif
 }
 EOF
-  $CC $CFLAGS $LDFLAGS $TMPDIR/$$.c -o $TMPDIR/$$.bin $opt #&> /dev/null
-  RET=$?
-  [ $RET -eq 0 ] && $TMPDIR/$$.bin
+  $CC $CFLAGS $LDFLAGS $TMPDIR/$$.c -o $TMPDIR/$$.bin $opt &> /dev/null
+  #$CC $CFLAGS $LDFLAGS $TMPDIR/$$.c -o $TMPDIR/$$.bin $opt #&> /dev/null
+  #RET=$?
+  #[ $RET -eq 0 ] && $TMPDIR/$$.bin
   RET=$?
   rm -f $TMPDIR/$$.{c,bin}
   return $RET
